LISA/LISAImplementation.cpp

```diff
@@ -241,14 +241,56 @@ class LISAImplementation : public Exchange::ILISA {
 
     virtual uint32_t Register(ILISA::INotification* notification) override
     {
+        // Make sure a callback is not registered multiple times.
+        ASSERT(std::find(_notificationCallbacks.begin(), _notificationCallbacks.end(), notification) == _notificationCallbacks.end());
+
+        _notificationCallbacks.push_back(notification);
+        notification->AddRef();
+
+        TRACE_GLOBAL(Trace::Information, (_T("Register INotification:%p"), notification));
+
         return Core::ERROR_NONE;
     }
 
     virtual uint32_t Unregister(ILISA::INotification* notification) override
     {
+        auto index(std::find(_notificationCallbacks.begin(), _notificationCallbacks.end(), notification));
+
+        // Make sure you do not unregister something you did not register !!!
+        ASSERT(index != _notificationCallbacks.end());
```

Comment:
Do we need assert if we are checking condition below?

Reply:
I followed this implementation of WebKit https://github.com/WebPlatformForEmbedded/ThunderNanoServices/blob/master/WebKitBrowser/WebKitImplementation.cpp#L631
It also looked a bit weird to me at first but I can find a benefit here. The check below is necessary anyway to make it work properly since there is no callback registered that was attempted to being remove. Additionally the assert will ring a bell during development that you're probably doing something wrong.